runtime/shibuya/src/lib.rs

```diff
@@ -1665,6 +1677,52 @@ type EventRecord = frame_system::EventRecord<
     <Runtime as frame_system::Config>::Hash,
 >;
 
+parameter_types! {
+    pub const EnterDuration: BlockNumber = 4 * HOURS;
+    pub const EnterDepositAmount: Balance = 2_000_000 * SBY;
+    pub const ExtendDuration: BlockNumber = 2 * HOURS;
+    pub const ExtendDepositAmount: Balance = 1_000_000 * SBY;
+    pub const ReleaseDelay: u32 = 2 * DAYS;
+}
+
+pub struct SafeModeWhitelistedCalls;
+impl Contains<RuntimeCall> for SafeModeWhitelistedCalls {
+    fn contains(call: &RuntimeCall) -> bool {
+        match call {
+            RuntimeCall::System(_) | RuntimeCall::SafeMode(_) | RuntimeCall::TxPause(_) => true,
+            _ => false,
+        }
+    }
+}
+
+impl pallet_safe_mode::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type Currency = Balances;
+    type RuntimeHoldReason = RuntimeHoldReason;
+    type WhitelistedCalls = SafeModeWhitelistedCalls;
+    type EnterDuration = EnterDuration;
+    type EnterDepositAmount = EnterDepositAmount;
+    type ExtendDuration = ExtendDuration;
+    type ExtendDepositAmount = ExtendDepositAmount;
+    type ForceEnterOrigin = EnsureRootWithSuccess<AccountId, ConstU32<9>>;
+    type ForceExtendOrigin = EnsureRootWithSuccess<AccountId, ConstU32<11>>;
+    type ForceExitOrigin = EnsureRoot<AccountId>;
+    type ForceDepositOrigin = EnsureRoot<AccountId>;
+    type ReleaseDelay = ReleaseDelay;
+    type Notify = ();
+    type WeightInfo = pallet_safe_mode::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_tx_pause::Config for Runtime {
+    type RuntimeCall = RuntimeCall;
+    type RuntimeEvent = RuntimeEvent;
+    type PauseOrigin = EnsureRoot<AccountId>; // Authority required to pause transactions.
+    type UnpauseOrigin = EnsureRoot<AccountId>; // Authority required to unpause transactions.
+    type WhitelistedCalls = ();
```

Comment:
In the similar way you did for the pallet_safe_mode, can you create a `TxPauseWhitelistedCalls` type?

An example (but you can check if other pallets can be whitelisted):

```rs
/// Calls that can bypass the tx-pause pallet.
/// We always allow system calls and timestamp since it is required for block production
pub struct TxPauseWhitelistedCalls;
impl Contains<pallet_tx_pause::RuntimeCallNameOf<Runtime>> for TxPauseWhitelistedCalls {
	fn contains(full_name: &pallet_tx_pause::RuntimeCallNameOf<Runtime>) -> bool {
		matches!(full_name.0.as_slice(), b"Sudo" | b"System" | b"Timestamp" | b"TxPause")
	}
}
```